src/components/App.js

```diff
@@ -1,216 +1,274 @@
+    await provider.send('eth_requestAccounts', []);
+    provider.on('network', (newNetwork, oldNetwork) => {
+      if (oldNetwork) {
+        window.location.reload();
+      }
+    }); // Reload on network change
     const signer = provider.getSigner(); //console.log(provider)
     let account = await signer.getAddress(); //console.log(account)
-    let netId = 5777
-    return [provider, signer, account, netId]
-  }
+    let netId = 5777;
+    return [provider, signer, account, netId];
+  }, []);
+
+  const loadBlockchainData = useCallback(async () => {
+    if (typeof window.ethereum !== 'undefined') {
+      let [provider, signer, account, netId] = await connectWallet(); //console.log(provider, signer, account, netId)
 
-  async loadBlockchainData(dispatch) {
-    if(typeof window.ethereum !== 'undefined'){
-      let [provider, signer, account, netId] = await this.connectWallet(); //console.log(provider, signer, account, netId)
-      try { // Load contracts
+      try {
+        // Load contracts
         const dChequeAddress = await dCheque.networks[netId].address;
         let dcheque = new ethers.Contract(dChequeAddress, dCheque.abi, signer);
-        window.dCheque = dcheque
+        window.dCheque = dcheque;
         const dChequeBalance = await provider.getBalance(dChequeAddress);
-        let userBalance = await dcheque.deposits(account)
-        this.setState({signer:signer, account: account, 
-                       dcheque: dcheque, dChequeAddress: dChequeAddress,
-                       dChequeBalance: ethers.utils.formatEther(dChequeBalance), 
-                       userBalance:ethers.utils.formatEther(userBalance)
-                      }) 
-      } catch (e){
-        console.log('error', e)
-        window.alert('Contracts not deployed to the current network')
+        let userBalance = await dcheque.deposits(account);
+        setState({
+          signer: signer,
+          account: account,
+          dcheque: dcheque,
+          dChequeAddress: dChequeAddress,
+          dChequeBalance: ethers.utils.formatEther(dChequeBalance),
+          userBalance: ethers.utils.formatEther(userBalance),
+        });
+        return true;
+      } catch (e) {
+        console.log('error', e);
+        window.alert('Contracts not deployed to the current network');
       }
-    } else{  //if MetaMask not exists push alert
-      window.alert('Please install MetaMask')
+    } else {
+      //if MetaMask not exists push alert
+      window.alert('Please install MetaMask');
     }
-  }
+  }, [connectWallet]);
 
-  constructor(props) {
-    super(props)
-    this.state = {
-      web3: 'undefined',
-      account: '',
-      dcheque: null,
-      dChequeAddress: null,
-      dChequeBalance: 0,
-      userBalance: 0
-    }
-  }
+  const updateReviewer = useCallback(() => {}, []);
+
+  useEffect(() => {
+    loadBlockchainData();
+  }, [loadBlockchainData]);
```

Comment:
remove dependency?

Reply:
by removing the dependency do you mean moving this into another file? I think we def should